src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java

```diff
@@ -705,13 +720,23 @@ public void deleteStack(final ItemStack stack)
      * @param state the block.
      * @return the float value.
      */
-    public float getStrVsBlock(final IBlockState state)
+    public float getStrVsBlock(final EnumHand hand, final IBlockState state)
     {
         float f = 1.0F;
 
-        if (!(this.mainInventory.get(this.currentItem)).isEmpty())
+        if (hand.equals(EnumHand.MAIN_HAND))
         {
-            f *= (this.mainInventory.get(this.currentItem)).getDestroySpeed(state);
+            if (!(this.mainInventory.get(this.mainItem)).isEmpty())
+            {
+                f *= (this.mainInventory.get(this.mainItem)).getDestroySpeed(state);
+            }
+        }
+        else if (hand.equals(EnumHand.OFF_HAND))
+        {
+            if (!(this.mainInventory.get(this.offhandItem)).isEmpty())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)